ogr/ogrsf_frmts/mssqlspatial/ogrmssqlspatialtablelayer.cpp

```diff
@@ -2217,9 +2217,6 @@ OGRErr OGRMSSQLSpatialTableLayer::CreateFeatureBCP(OGRFeature *poFeature)
                         {
                         }
                     }
-                    if (Failed2(bcp_moretext(hDBCBCP, 0, nullptr)))
```

Comment:
I'm not familiar at all with bcp_moretext(), but removing it here seems to break the "symmetry" with the OFTString and geometry cases above. Any reason why removing it is needed here, and not in those cases ?
Perhaps add in your test case writing a feature with a binary field that is set, but of size 0, to see if there isn't something weird happening (and perhaps also adding another dynamic field, let's say of type String after the binary field, to check that there isn't any mismatch in the bcp batch)

Reply:
done, and you were right: it must be called when size is 0.